handlers/playback.go

```diff
@@ -0,0 +1,101 @@
+package handlers
+
+import (
+	"errors"
+	"io"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/livepeer/catalyst-api/config"
+	catErrs "github.com/livepeer/catalyst-api/errors"
+	"github.com/livepeer/catalyst-api/log"
+	"github.com/livepeer/catalyst-api/playback"
+)
+
+func ManifestHandler() httprouter.Handle {
+	return func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		var requestID = config.RandomTrailer(8)
+
+		err := req.ParseForm()
+		if err != nil {
+			handleError(err, req, requestID, w)
+			return
+		}
+
+		key := req.Form.Get(playback.ManifestKeyParam)
+		if err := checkKey(key); err != nil {
+			handleError(err, req, requestID, w)
+			return
+		}
+		manifest, err := playback.Manifest(playback.PlaybackRequest{
```

Comment:
I am not sure how go behave, is this reader returned from the Manifest function getting closed here? Or do we need to return a `io.ReadCloser` instead and defer here manifest.Close()?

Reply:
we're ok because the underlying object is a bytes.Buffer which only implements reader: https://github.com/grafov/m3u8/blob/master/writer.go#L73